config/initializers/rack_attack.rb

```diff
@@ -17,32 +17,59 @@ class Rack::Attack
     "/api/docs",
   ].freeze
 
+  def self.api_request?(request)
+    request.path.starts_with?("/api/")
+  end
+
+  def self.public_api_path?(request)
+    PUBLIC_API_PATH_PREFIXES.any? { |prefix| request.path.starts_with?(prefix) }
+  end
+
+  def self.protected_path?(request)
+    PROTECTED_ROUTES.include?(request.path)
+  end
+
+  def self.csp_report_path?(request)
+    request.path == "/csp_reports"
+  end
+
+  def self.get_an_identity_webhook_path?(request)
+    request.path.starts_with?("/api/v1/get_an_identity/webhook_messages")
+  end
+
+  def self.auth_token(request)
+    request.get_header("HTTP_AUTHORIZATION")
+  end
+
   # Throttle protected routes by IP (5rpm)
   throttle("protected routes (hitting external services)", limit: 5, period: 1.minute) do |request|
-    request.ip if PROTECTED_ROUTES.include?(request.path)
+    request.ip if protected_path?(request)
   end
 
   # Throttle /csp_reports requests by IP (5rpm)
   throttle("csp_reports req/ip", limit: 5, period: 1.minute) do |request|
-    request.ip if request.path == "/csp_reports"
+    request.ip if csp_report_path?(request)
   end
 
   # Throttle /api/v1/get_an_identity/webhook_messages requests by IP (1000 requests per 5 minutes)
   throttle("API get an identity webhook message requests by ip", limit: 1000, period: 5.minutes) do |request|
-    request.ip if request.path.starts_with?("/api/v1/get_an_identity/webhook_messages")
+    request.ip if get_an_identity_webhook_path?(request)
   end
 
-  # Throttle /api requests by auth token (1000 requests per 5 minutes)
+  # Throttle private /api requests by auth token (1000 requests per 5 minutes)
   throttle("API requests by auth token", limit: 1000, period: 5.minutes) do |request|
-    public_api_path = PUBLIC_API_PATH_PREFIXES.any? { |prefix| request.path.starts_with?(prefix) }
+    auth_token(request) if api_request?(request) && !public_api_path?(request)
+  end
 
-    if request.path.starts_with?("/api/") && !public_api_path
-      request.get_header("HTTP_AUTHORIZATION")
-    end
+  # Throttle non-api requests (300 requests per 5 minutes)
+  throttle("non-API requests by ip", limit: 300, period: 5.minutes) do |request|
+    request.ip unless api_request?(request)
```

Comment:
should we also ensure `public_api_path` is included here?

Reply:
We could do, might be worth having a separate, explicit throttle for them or as it stands they just fallback to the catchall

Reply:
Pushed a commit with an explicit rate limit for them